app/src/main/java/app/revanced/integrations/youtube/patches/components/KeywordContentFilter.java

```diff
@@ -0,0 +1,156 @@
+package app.revanced.integrations.youtube.patches.components;
+
+import static app.revanced.integrations.shared.StringRef.str;
+import static app.revanced.integrations.youtube.ByteTrieSearch.convertStringsToBytes;
+
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.ByteTrieSearch;
+import app.revanced.integrations.youtube.settings.Settings;
+
+/**
+ * <pre>
+ * Allows hiding home feed and search results based on keywords and/or channel names.
+ *
+ * Limitations:
+ * - Searching for a keyword phrase will give no search results
+ * - Filtering a channel name can still show Shorts from that channel in the search results
+ * - Some layout component residue will remain, such as the video chapter previews for some search results.
+ * - Keywords are case sensitive, but some casing variation is manually added.
+ *   (ie: "mr beast" automatically filters "Mr Beast" and "MR BEAST").
+ */
+@SuppressWarnings("unused")
+@RequiresApi(api = Build.VERSION_CODES.N)
+final class KeywordContentFilter extends Filter {
+
+    /**
+     * Minimum keyword/phrase length to prevent excessively broad content filtering.
+     */
+    private static final int MINIMUM_KEYWORD_LENGTH = 3;
+
+    /**
+     * Substrings that are always first in the path.
+     */
+    private final StringFilterGroup startsWithFilter = new StringFilterGroup(
```

Comment:
Exposing this filter group to the user in an input preference may be a good idea. By default the current values can be added, but in some cases users may want to remove or add their own filters.

Reply:
> But Shorts use the same layout paths for both home and search, which makes it difficult to selectively turn on/off like this.

I think edge cases like this can be left unconsidered, but the most likely scenarios such as the one you mentioned [here](https://github.com/ReVanced/revanced-integrations/pull/584#discussion_r1519512103) could be turned into a toggleable switch.